tests/0113-cooperative_rebalance.cpp

```diff
@@ -1711,25 +1709,45 @@ static void n_wildcard () {
       Test::Say("Creating two topics with 2 partitions each that match regex\n");
       test_create_topic(NULL, topic_name_1.c_str(), 2, 1);
       test_create_topic(NULL, topic_name_2.c_str(), 2, 1);
-      test_wait_topic_exists(c1->c_ptr(), topic_name_1.c_str(), 10*1000);
-      test_wait_topic_exists(c1->c_ptr(), topic_name_2.c_str(), 10*1000);
+      /* The consumers should autonomously discover these topics and start
+       * consuming from them. This happens in the background - is not
+       * influenced by whether we wait for the topics to be created before
+       * continuing the main loop. It is possible that both topics are
+       * discovered simultaneously, requiring a single rebalance OR that
+       * topic 1 is discovered first (it was created first), a rebalance
+       * initiated, then topic 2 discovered, then another rebalance
+       * initiated to include it.
+       */
       created_topics = true;
     }
 
     if (Test::assignment_partition_count(c1, NULL) == 2 &&
         Test::assignment_partition_count(c2, NULL) == 2 && !deleted_topic) {
-      TEST_ASSERT(rebalance_cb1.nonempty_assign_call_cnt == 1,
-                  "Expecting C_1's nonempty_assign_call_cnt to be 1 not %d ",
-                  rebalance_cb1.nonempty_assign_call_cnt);
-      TEST_ASSERT(rebalance_cb2.nonempty_assign_call_cnt == 1,
-                  "Expecting C_2's nonempty_assign_call_cnt to be 1 not %d ",
-                  rebalance_cb2.nonempty_assign_call_cnt);
-      TEST_ASSERT(rebalance_cb1.nonempty_revoke_call_cnt == 0,
-                  "Expecting C_1's nonempty_revoke_call_cnt to be 0 not %d ",
-                  rebalance_cb1.nonempty_assign_call_cnt);
-      TEST_ASSERT(rebalance_cb2.nonempty_revoke_call_cnt == 0,
-                  "Expecting C_2's nonempty_revoke_call_cnt to be 0 not %d ",
-                  rebalance_cb2.nonempty_assign_call_cnt);
+
+      if (rebalance_cb1.nonempty_assign_call_cnt == 1) {
+        /* just one rebalance was required */
+        TEST_ASSERT(rebalance_cb1.nonempty_assign_call_cnt == 1,
+                    "Expecting C_1's nonempty_assign_call_cnt to be 1 not %d ",
+                    rebalance_cb1.nonempty_assign_call_cnt);
+        TEST_ASSERT(rebalance_cb2.nonempty_assign_call_cnt == 1,
+                    "Expecting C_2's nonempty_assign_call_cnt to be 1 not %d ",
+                    rebalance_cb2.nonempty_assign_call_cnt);
+      } else {
+        /* two rebalances were required (occurs infrequently) */
```

Comment:
Why not just TEST_ASSERT( call_cnt == 1 .. or 2) ?

Reply:
both C_1 and C_2 's count must either be 1 or both must be 2. i could write this differently, but i think spending any effort beyond whatever i first wrote out to articulate this is well below the bar of we-should-be-caring-about-it. it's easy to read, it's not in a key part of the code etc.